### PR TITLE
Initial commit of BigtableToParquet

### PR DESCRIPTION
**Pull request by YadnikiPawar**
Adding BigtableToParquet

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,6 +95,7 @@
     <cassandra.driver.version>3.6.0</cassandra.driver.version>
     <kms.version>1.20.0</kms.version>
     <proto-kms.version>0.67.0</proto-kms.version>
+    <parquet.version>2.14.0</parquet.version>
```
Comment (sabhyankar) [inline pom.xml]: This should be removed. Please use ${beam.version} for the beam-sdks-java-io-parquet depdendency.

File: pom.xml [hunk 2]
```diff
@@ -445,6 +446,12 @@
       <artifactId>avro</artifactId>
       <version>${avro.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-sdks-java-io-parquet</artifactId>
+      <version>${parquet.version}</version>
```
Comment (sabhyankar) [inline pom.xml]: beam.version

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    /**
+     * Steps:
+     * 1) Read records from Bigtable.
+     * 2) Convert a Bigtable Row to a GenericRecord.
+     * 3) Write GenericRecord(s) to GCS in parquet format.
+     */
+    pipeline
+        .apply("Read from Bigtable", read)
+        .apply("Transform to Parquet", MapElements.via(new BigtableToParquetFn()))
+        .setCoder(AvroCoder.of(GenericRecord.class, BigtableRow.getClassSchema()))
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Can you indent/reformat this line?

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        .apply(
+            "Write to Parquet in GCS",
+            FileIO.<GenericRecord>write()
+                .via(ParquetIO.sink(BigtableRow.getClassSchema()))
+                .to(options.getOutputDirectory())
+                .withPrefix(options.getFilenamePrefix())
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: As a test, can you create a template and then run the template by passing a specific --filenamePrefix=my-output

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+   */
+  private static byte[] toByteArray(final ByteString byteString) {
+    try {
+      ZeroCopyByteOutput byteOutput = new ZeroCopyByteOutput();
+      UnsafeByteOperations.unsafeWriteTo(byteString, byteOutput);
+      return byteOutput.bytes;
+    } catch (IOException e) {
+      return byteString.toByteArray();
+    }
+  }
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Why dont you simply make this method in BigtableToAvro.java as protected so you can reuse without duplicating?

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+  private static final class ZeroCopyByteOutput extends ByteOutput {
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Same for this.

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 5 above)
Comment (YadnikiPawar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Done

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 4 above)
Comment (YadnikiPawar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: @sabhyankar 
Create a template as in? 
Cause this is a legacy template.

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 6 above)
Comment (YadnikiPawar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: @sabhyankar  Don't need this one to be protected. Its used only within the previous function

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 4 above)
Comment (sabhyankar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Yes - Create a legacy template and then call gcloud dataflow jobs run with that template and pass fileNamePrefix=my-output as a parameter.

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 6 above)
Comment (sabhyankar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Got it - Thanks.

File: src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java (hunk 4 above)
Comment (YadnikiPawar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: Okay got it

Comment (YadnikiPawar) [reply src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java]: @sabhyankar 
It works fine.

**Comment by YadnikiPawar** [comment]
@sabhyankar  Changes are added and the PR is ready

**Pull request merged**